pkg/solver/matcher/matcher.go

```diff
@@ -101,27 +93,30 @@ func GetMatchingDeals(
 			}
 			matchSpan.AddEvent("db.get_match_decision.done")
 
```

Comment:
A potential optimization for later but on line 88 we are calling `db.GetMatchDecision(resourceOffer.ID, jobOffer.ID)` in each step of the loop.  Perhaps we can change this to instead get called as `GetMatchDecisions()` and have it hoisted just before the loop starts.  It's more to save on multiple network calls to the DB and have the code sift through the decisions but I think it should speed it up a bit assuming we have the indexes on the DB properly lined up

Reply:
Opened an issue and referenced this optimization for future work: https://github.com/Lilypad-Tech/lilypad/issues/503